cmd/litcli/main.go

```diff
@@ -244,3 +295,69 @@ func printRespJSON(resp proto.Message) { // nolint
 
 	fmt.Println(string(jsonBytes))
 }
+
+func connectTapdClient(ctx *cli.Context) (grpc.ClientConnInterface, func(),
+	error) {
+
+	lndConn, cleanup, err := connectClient(ctx, false)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error connecting client: %w", err)
+	}
+
+	defer cleanup()
+
+	var suffixBytes [4]byte
+	superMacID := session.NewSuperMacaroonRootKeyID(suffixBytes)
+	lndClient := lnrpc.NewLightningClient(lndConn)
+	ctxb := context.Background()
+
+	tapdPerms := perms.RequiredPermissions
+
+	deDupMap := make(map[string]map[string]bool)
+	for _, methodPerms := range tapdPerms {
+		for _, methodPerm := range methodPerms {
+			if methodPerm.Action == "" || methodPerm.Entity == "" {
+				continue
+			}
+
+			if deDupMap[methodPerm.Entity] == nil {
+				deDupMap[methodPerm.Entity] = make(
+					map[string]bool,
+				)
+			}
+			deDupMap[methodPerm.Entity][methodPerm.Action] = true
+		}
+	}
+
+	rpcPerms := make([]*lnrpc.MacaroonPermission, 0, len(tapdPerms))
+	rpcPerms = append(rpcPerms, &lnrpc.MacaroonPermission{
+		Entity: "uri",
+		Action: "/lnrpc.Lightning/CheckMacaroonPermissions",
+	})
+	for entity, actions := range deDupMap {
+		for action := range actions {
+			rpcPerms = append(rpcPerms, &lnrpc.MacaroonPermission{
+				Entity: entity,
+				Action: action,
+			})
+		}
+	}
+
+	macResp, err := lndClient.BakeMacaroon(
+		ctxb, &lnrpc.BakeMacaroonRequest{
+			Permissions:              rpcPerms,
+			RootKeyId:                superMacID,
+			AllowExternalPermissions: true,
+		},
```

Comment:
i wonder if we can't rather expand on the new BakeSuperMacaroon functionality?

Reply:
we can just edit the proto message such that it takes an optional sub-server name

Reply:
Ah, cool, baking a super macaroon works well here. Sure, we could narrow the permissions later, but for now this is great.